model/riscv_sys_control.sail

```diff
@@ -278,7 +278,6 @@ function dispatchInterrupt(priv : Privilege) -> option((InterruptType, Privilege
    * Absence of U-mode implies absence of S-mode.
```

Comment:
I think the comment is a bit wrong and the check for S mode is redundant. Should it not be something like this?

```
// If we don't have S or N then interrupts cannot be delegated and must
// always be handled in machine mode.
if not(extensionEnabled(Ext_S)) & not(extensionEnabled(Ext_N) then {
  let enable_pending...
```

Removing the assertion looks correct to me. I would guess it was correct when written, and the not removed when N was added.

Reply:
Yeah, that was my interpretation as well. I'll reword the comment.

Reply:
The title of this PR is no longer accurate, but I think that has tidied up the function a lot.